### DISCUSSION TITLE
Restart stream instead of exit

### DISCUSSION THREAD

**Discussion by Trung0246** [streamlink/streamlink#3671] [Q&A]
Referring to https://github.com/streamlink/streamlink/issues/3490: is there's a way to configure streamlink such that it will attempt to restart the stream instead bomb out? Or do I have to manually run the command again everytime it bomb out?

**Comment by WolfganP** [comment]
You may try running streamlink with `--player-external-http` flags (https://streamlink.github.io/cli.html) and the player launched manually with the proper retry commands.

**Comment by Trung0246** [comment]
Hmm I saw `--player-continuous-http` too, is there's a difference between them? Apparently `--player-external-http` will just print out IP URLs and it's painful to copy and paste them into "Network Stream" (unless streamlink let me pipe those URLs into VLC).

**Comment by Trung0246** [comment]
Resurrecting the discussion since the issue is back when using `--player-external-http`, no idea why I keep getting this or my internet is just unstable as streamlink should restart stream when I use the flag but apparently it's not?

```
[cli][info] Found matching plugin twitch for URL https://twitch.tv/veibae
[cli][info] Available streams: audio_only, 160p (worst), 360p, 480p, 720p, 720p60, 1080p60 (best)
[cli][info] Starting server, access with one of:
[cli][info]  http://127.0.0.1:59542/
[cli][info]  http://<personal ipv4?>:59542/
[cli][info]  http://<personal ipv4?>:59542/
[cli][info]  http://<personal ipv4?>:59542/
[cli][info]  http://<personal ipv4?>:59542/
[cli][info] Got HTTP request from VLC/3.0.12 LibVLC/3.0.12
[cli][info] Opening stream: 1080p60 (hls)
[plugins.twitch][info] Will skip ad segments
[plugins.twitch][info] Waiting for pre-roll ads to finish, be patient
[stream.hls][info] Filtering out segments and pausing stream output
[stream.hls][info] Resuming stream output
error: Error when reading from stream: Read timeout, exiting
Exception in thread Thread-TwitchHLSStreamWriter:
[cli][info] Stream ended
Traceback (most recent call last):
[cli][info] Closing currently open stream...
  File "C:\Program Files (Portable & CMD)\streamlink\pkgs\urllib3\response.py", line 438, in _error_catcher
    yield
  File "C:\Program Files (Portable & CMD)\streamlink\pkgs\urllib3\response.py", line 519, in read
```

Trying to run streamlink with `-l debug` enabled now...
